src/util/std_code.h

```diff
@@ -196,6 +196,12 @@ class code_assignt:public codet
     copy_to_operands(lhs, rhs);
   }
 
+  code_assignt(const exprt &lhs, const exprt &rhs, const source_locationt &loc):
```

Comment:
Having this constructor allows me to declare a `code_assignt` as `const` which I couldn't do before because `source_location` needs to be set. 
There are other places where we could use this constructor, to have more succinct code and make variables `const`, but I don't think it is very important.

Reply:
Don't do this here. I've got a plan to make this for all `exprt`, which I'll propose in a separate PR.